http2_general/test_h2_stream_states.py

```diff
@@ -0,0 +1,57 @@
+"""Functional tests for stream states."""
+
+__author__ = "Tempesta Technologies, Inc."
+__copyright__ = "Copyright (C) 2023 Tempesta Technologies, Inc."
+__license__ = "GPL2"
+
+from hyperframe.frame import PriorityFrame, RstStreamFrame, WindowUpdateFrame
+
+from http2_general.helpers import H2Base
+
+
+class TestClosedStreamState(H2Base):
+    def test_rst_stream_frame_in_closed_state(self):
+        """
+        An endpoint that sends a frame with the END_STREAM flag set or a RST_STREAM frame might
+        receive a WINDOW_UPDATE or RST_STREAM frame from its peer in the time before the peer
+        receives and processes the frame that closes the stream.
+        RFC 9113 5.1
+
+        Tempesta MUST not close connection.
+        """
+        self.start_all_services()
+
+        client = self.get_client("deproxy")
+        client.send_request(request=self.post_request, expected_status_code="200")
+        client.send_bytes(RstStreamFrame(stream_id=1).serialize())
+        client.send_request(request=self.post_request, expected_status_code="200")
+
+    def test_window_update_frame_in_closed_state(self):
+        """
+        An endpoint that sends a frame with the END_STREAM flag set or a RST_STREAM frame might
+        receive a WINDOW_UPDATE or RST_STREAM frame from its peer in the time before the peer
+        receives and processes the frame that closes the stream.
+        RFC 9113 5.1
+
+        Tempesta MUST not close connection.
+        """
+        self.start_all_services()
+
+        client = self.get_client("deproxy")
+        client.send_request(request=self.post_request, expected_status_code="200")
+        client.send_bytes(WindowUpdateFrame(stream_id=1).serialize())
```

Comment:
It's strange that we send WindowUpdateFrame on a closed stream and it is ok. In the test below you write that "An endpoint MUST NOT send frames other than PRIORITY on a closed stream."

Reply:
Tempesta can receive WINDOW_UPDATE or RST_STREAM in closed stream, because client can send these frames before receiving END_STREAM. We just must not close connection or stream.

> Also i think, that we should implement tests for other stream states, but we can do it in separate PR

I added tests that are not in [h2spec](https://github.com/summerwind/h2spec/blob/master/http2/5_1_stream_states.go).